modules/videoio/cmake/init.cmake

```diff
@@ -0,0 +1,35 @@
+macro(add_backend backend_id cond_var)
+  if(${cond_var})
+    include("${CMAKE_CURRENT_LIST_DIR}/detect_${backend_id}.cmake")
+  endif()
+endmacro()
+
+macro(add_3p_target name inc link def)
+  add_library(ocv.3rdparty.${name} INTERFACE)
```

Comment:
1.
Please move this macro into common `OpenCVUtils.cmake` (this logic can be re-used for imgcodecs - if `BUILD_xxx=OFF`).

2.
Lets change name:
- add `ocv_` prefix
- it should be used for "external" targets
- something like this: `ocv_add_external_dependency_target()`

3.
In `BUILD_SHARED_LIBS=ON` configuration we should not export extra targets in OpenCVModules.cmake, because they are not used. Lets mark these targets for this configuration as `IMPORTED`.

Reply:
I don't want to extract it to OpenCVUtils.cmake yet because it has very basic interface which will be changed later.